Common/cpp/SharedItems/ShareableValue.cpp

```diff
@@ -136,6 +140,17 @@ void ShareableValue::adapt(jsi::Runtime &rt, const jsi::Value &value, ValueType
   }
 }
 
+void cleanupShareable(ShareableValue &sv) {
```

Comment:
The function doesn't make sense at all. If an object is the last reference to the function then the function will garbage collected even without the function. If it's not the case then you will remove the function even if it's needed somewhere else.

Reply:
It's not that it doesn't make sense at all, I was although sceptic about it. Well, I checked and it does introduce some problems so it's not the best solution but it solves a problem which would persist without it.

Reply:
It in fact introduces errors but covers a leak that will persist without it

Reply:
It still doesn't make sense.

Reply:
Which cases do you think it covers?

Reply:
discussed internally